### ISSUE TITLE
July 10, 2024 ASF Board Report

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#10282]
### Is your feature request related to a problem or challenge?

Per https://www.apache.org/foundation/board/reporting, for the first three months of a project it should submit monthly board reports to the ASF board

Subsequently, per  https://whimsy.apache.org/roster/committee/datafusion the DataFusion ASF board report schedule is

March, June, September, December

### Describe the solution you'd like

I would like to draft a board report for the ASF board meeting, ideally with community help. 

The meetings are typically in the second or third week of the month

### Describe alternatives you've considered

I plan to do this in the same style that worked well in Arrow (see an example from @andygrove 
 here https://lists.apache.org/thread/7w4mgy98qomc6drvj2fo81gvhq6p0boc) -- make a google doc (or issue) that people can add relevant content to and then the chair (me for the time being) submits it to the board

**Comment by alamb** [comment]
Here is the reminder email

```
---------- Forwarded message ---------
From: Shane Curcuru <board-chair@apache.org>
Date: Mon, Jul 1, 2024 at 5:20 PM
Subject: DataFusion Board Report due by Wed Jul 10th - Initial Reminder
To: Andrew Lamb <alamb@apache.org>
Cc: <private@datafusion.apache.org>


Hello,

According to our records, you are listed as the chair of DataFusion,
a committee that is due to submit a report by Wed Jul 10th
for the next ASF board meeting. This is an initial reminder to prepare a
report for DataFusion and submit it as described below.

In the Apache governance model, the ASF board delegates responsibility for
managing projects to PMCs. This allows projects to govern themselves, in
terms of their own development goals, guidelines, and volunteer spirit,
within the scope of our purpose as an open source foundation. The state
allows us to provide an umbrella of corporate protection to our projects
and volunteers, but only to the extent that we retain active and effective
oversight of each project's operation on behalf of the public's interest.

To enable the board to provide oversight across the foundation, each PMC
is tasked with providing the board a quarterly report on the health of
their project. This allows us to hear your heartbeat, to see the project
through your eyes, and to inform the public through our meeting minutes.

Thank you for being a responsible project chair and helping us maintain
oversight over the Apache Software Foundation. If, for whatever reason,
a full report is not possible by the deadline, please report just that.
It's okay to postpone a report by a month.

Please note that we will be reviewing and approving these reports,
asynchronously, during the week prior to our scheduled board meeting.
Reports received after Wed Jul 10th will be postponed to the
next regular meeting. Please try to submit your report early.


Submitting your report
----------------------

Full details about reporting to the board are at

  https://www.apache.org/foundation/board/reporting

Please be aware that the board is looking for your personal observations,
assessment, and ideas, not just raw statistics.

Chairs may use one of several mechanisms to submit or edit their report:

 a) the Apache Reporter Service
    https://reporter.apache.org/

 b) the Whimsy online agenda tool
    https://whimsy.apache.org/board/agenda/2024-07-17/DataFusion

 c) carefully editing and committing changes to the dated agenda in
    https://svn.apache.org/repos/private/foundation/board

 d) or, if none of the above work, send an email to [board@apache.org](mailto:board@apache.org) with
    Subject: [REPORT] DataFusion

If you believe it won't be possible to prepare a report before the deadline,
or if the PMC is aware that the Chair is unavailable, please report that
and we can reschedule or have someone else report on your behalf.
```

**Comment by alamb** [comment]
Here is a draft for anyones comments / review: https://docs.google.com/document/d/1lV-cFZGHCSrTiaLW1gyEMDKW-9nf47UW8xK19QCqbVk/edit

**Comment by alamb** [comment]
Kudos to @andygrove @phillipleblanc @lukekim and @Omega359  for the help

I have submitted the report. Here is the final content

```
2024-07-10 DataFusion ASF Board Report 
https://github.com/apache/datafusion/issues/10282 

DataFusion PMC Chair Note: Please add any relevant comments / content to this document. I (Andrew Lamb) will submit to the ASF board on Wed July 10, 2024 (about one week prior to the scheduled board meeting). 

New projects submit reports every month for the first three months. This is our last of those three monthly reports. Our next one is due on July 10, 2024

The format of this report and the metrics are from https://reporter.apache.org/wizard/?datafusion 

The rationale and process for this report: https://www.apache.org/foundation/board/reporting  
Past examples: [2024-06-12 DataFusion ASF Board Report](https://docs.google.com/document/d/1h4yjvomQO0XdzxKuE4aBSWGNliFFmn8GADd8DlPuXBw/edit)



## Description:
The mission of Apache DataFusion is the creation and maintenance of software 
related to an extensible query engine

## Project Status:
Current project status: New + Ongoing (high activity)
Issues for the board: None


## Membership Data:
Apache DataFusion was founded 2024-04-16 (3 months ago)
There are currently 33 committers and 13 PMC members in this project.
The Committer-to-PMC ratio is roughly 9:4.

Community changes, past month:
- Mehmet Ozan Kabak was added to the PMC on 2024-06-12
- Ruihang Xia was added to the PMC on 2024-06-12
- Lewis Zhang was added as committer on 2024-06-14


## Project Activity:

The project continues to be quite active with many PRs and issues opened and
closed per day.

We started working on a project blog [1] (previously we used the arrow blog)
and hope to have our first blog post as an independent project later this
month.

There was a well attended face to face meetup in San Francisco, CA USA in June
[2]. We have one planned for Hangzhou, China in July[3]. There appears
 significant interest in these events and there are at least 2 more planned
 for September in New York, NY USA and in Belgrade, Serbia

The community around DataFusion is growing too. For example, Spice AI has made
an initial contribution of TableProviders to datafusion-contrib [4] for
PostgreSQL, MySQL, DuckDB, and SQLite, enabling these data sources to be
easily queried through DataFusion.

[1]: https://datafusion.apache.org/blog/
[2]: https://github.com/apache/datafusion/discussions/10800
[3]: https://github.com/apache/datafusion/discussions/10341
#discussioncomment-9738748
[4]: https://github.com/datafusion-contrib/datafusion-table-providers

### DataFusion core
https://github.com/apache/datafusion

We released version 39.0.0, continuing our schedule of monthly releases and
are on track to release version 40.0.0 in the next day or two.

Some projects we have been working on recently involve adding support for more
flexible use of Parquet files including indexing and extracting statistics. We
are also working with the community to make extending SQL planning[2] easier
and extending file format support[3], as well as fixing bugs found with a SQL
fuzzer[4], and improving performance with StringView[5].

It has been nice to see several good examples of cross contributor/company
collaboration such as [6] and [7].

We have also been making external presentations[1]

[1]: https://github.com/apache/datafusion/issues/10969
[2]: https://github.com/apache/datafusion/issues/10534
[3]: https://github.com/apache/datafusion/pull/11060
[4]: https://github.com/apache/datafusion/issues/11030
[5]: https://github.com/apache/datafusion/issues/10918
[6]: https://github.com/apache/datafusion/pull/11203
[7]: https://github.com/apache/datafusion/issues/10534

### Sub project: DataFusion Python

https://github.com/apache/datafusion-python

The DataFusion Python project continues to receive updates as new versions of
the core DataFusion project are released. There have also been some minor
improvements to improve user experience.


### Sub project: DataFusion Comet

https://github.com/apache/datafusion-comet

The Comet project is very active and is working towards an initial 0.1.0
source release. Initial benchmark results were published to
https://datafusion.apache.org/comet/contributor-guide/benchmarking.html.


### Sub project: DataFusion Ballista
https://github.com/apache/datafusion-ballista
https://github.com/apache/datafusion-ballista-python

The Ballista subproject is not very actively maintained, but there have been
some contributions recently to upgrade to more recent versions of the core
DataFusion project.

### Recent Releases
* PYTHON-39.0.0 was released on 2024-07-02.
* 39.0.0 was released on 2024-06-10.
* PYTHON-38.0.1 was released on 2024-05-30.
* PYTHON-37.1.0 was released on 2024-05-13.
* 38.0.0 was released on 2024-05-10.


## Community Health:
Community health is good -- we recently hit the 600 total contributors mark according to Github. This number is partially inflated from initially being part of the Arrow mono repo but the trend is healthy non the less.

It is hard to keep track of everything going on these days, which is a good thing. While it is always a struggle to get enough code review, the committers keep things going and the community helps each other out with reviews.
```
